multinode_integration_tests/src/main.rs

```diff
@@ -342,6 +349,104 @@ impl MockNode {
     }
 }
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+struct ProbeTarget {
+    ip_address: IpAddr,
+    port_specs: Vec<PortSpec>,
+}
+
+impl FromStr for ProbeTarget {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut main_pieces = s.split(':').collect_vec();
+        if main_pieces.len() != 2 {
+            return Err("Syntax: <IP address>:<U|T><port>/<U|T><port>/...".to_string());
+        }
+        let ip_address_string = main_pieces.remove(0);
+        let port_specs_string = main_pieces.remove(0);
+        let ip_address = match IpAddr::from_str(ip_address_string) {
+            Ok(ip_addr) => ip_addr,
+            Err(_) => return Err("Syntax: <IP address>:<U|T><port>/<U|T><port>/...".to_string()),
+        };
+        let port_spec_strings = port_specs_string.split('/').collect_vec();
+        let port_spec_results = port_spec_strings
+            .into_iter()
+            .map(PortSpec::from_str)
+            .collect_vec();
+        let first_error_opt = port_spec_results
+            .iter()
+            .find(|result| result.is_err())
+            .map(|err| err.clone().err().unwrap());
```

Comment:
Because of the previously depicted quastionability about the types, I cannot make up my mind about if the `.clone()` belongs there and why. It might be possible to factor it out but it depends on the type...yeah.

Reply:
This has become moot.